docs/examples/grpc/README.md

```diff
@@ -1,91 +1,166 @@
-   at least
-   [quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.30.0](https://quay.io/kubernetes-ingress-controller/nginx-ingress-controller)
-   for grpc support.
-4. You have a backend application running a gRPC server and listening for TCP
-   traffic.  If you prefer, you can use the
-   [fortune-teller](https://github.com/kubernetes/ingress-nginx/tree/master/images/grpc-fortune-teller)
-   application provided here as an example.
-
-### Step 1: kubernetes `Deployment`
+2. You have a domain name such as `example.com` that is configured to route traffic to the ingress controller. You're also responsible for provisioning an SSL certificate for the ingress).
+3. You have the nginx-ingress controller installed as per docs for gRPC support.
+4. You have a backend application running a gRPC server and listening for TCP traffic.  If you want, you can use https://raw.githubusercontent.com/grpc/grpc-go/master/examples/features/reflection/server/main.go as an example.
 
+### Step 1: Create kubernetes `Deployment` of gRPC app
+
+- Make sure your gRPC application pod is running and listening for connections. For example you can try a kubectl command like this below ;
 ```sh
-$ kubectl create -f app.yaml
+$ kubectl get po -o wide | grep go-grpc-greeter-server
 ```
+- If you have a gRPC app deployed in your cluster, then skip this section and continue from Step 2 below. 
 
-This is a standard kubernetes deployment object.  It is running a grpc service
-listening on port `50051`.
+  - As an example grpc application, we use this https://raw.githubusercontent.com/grpc/grpc-go/master/examples/features/reflection/server/main.go . To create a container image out of this app, a Dockerfile is shown below. You can create a image named `go-grpc-greeter-server` using this Dockerfile ;
 
-The sample application
-[fortune-teller-app](https://github.com/kubernetes/ingress-nginx/tree/master/images/grpc-fortune-teller)
-is a grpc server implemented in go. Here's the stripped-down implementation:
+```
+FROM golang as build
 
-```go
-func main() {
-	grpcServer := grpc.NewServer()
-	fortune.RegisterFortuneTellerServer(grpcServer, &FortuneTeller{})
-	lis, _ := net.Listen("tcp", ":50051")
-	grpcServer.Serve(lis)
-}
+ADD https://github.com/protocolbuffers/protobuf/releases/download/v3.15.8/protoc-3.15.8-linux-x86_64.zip /home/downloads/protoc/
+
+RUN apt update && \
+  apt install -y unzip && \
+  cd /home/downloads/protoc && \
+  unzip protoc-3.15.8-linux-x86_64.zip && \
+  cp bin/protoc /usr/bin/protoc && \
+  go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.26 && \
+  go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1 && \
+  cd /go/src && \
+  git clone https://github.com/grpc/grpc-go && \
+  cd /go/src/grpc-go/examples/features/reflection/server && \
+  go build -o /greeter-server main.go
+
+
+FROM gcr.io/distroless/base-debian10
+
+COPY --from=build /greeter-server /
+
+EXPOSE 50051
+
+CMD ["/greeter-server"]
+
+```
```

Comment:
Can we change/replace this snippet to a link to the Dockerfile?

Reply:
I removed the snippet and put a link there like https://github.com/kubernetes/ingress-nginx/blob/master/docs/examples/grpc/Dockerfile . But this link will be valid only after/if this PR merges, because the Dockerfile is currently only in my fork . I don't know how to create a relative link from the README to the Dockerfile that is in the same subdir as the README.md. If relative link is better, please help me to know how to put relative link.

Reply:
I think you can do a 
```
[Dockerfile]
```
and this becames relative

Anyway, you can test with "make live-docs" on root directory